src/main/resources/config/liquibase/changelog/20220602230300_changelog.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-3.8.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
+    <changeSet author="stoehrj" id="20220602230300">
+        <addColumn tableName="result">
+            <column name="test_case_amount" type="tinyint" defaultValue="0"/>
+        </addColumn>
+        <addColumn tableName="result">
+            <column name="passed_test_case_amount" type="tinyint" defaultValue="0"/>
+        </addColumn>
+        <addColumn tableName="result">
+            <column name="code_issue_amount" type="tinyint" defaultValue="0"/>
```

Comment:
In practice, having more than 127 test cases should not really be the case for exercises. However, I think for big exercises with a lot of code there could be more code issues in a submission (e.g., Checkstyle warning on tabs vs. spaces and the same error being repeated many times). In that case it would probably result in some weird and hard-to-debug values (or does Hibernate refuse to write results in that case?).

I think that the additional three bytes needed by upgrading to a `smallint` for those three values should not be a problem, but definitely eliminate that potential source of error.

Reply:
This was a request by Stephan to keep the required disk space of a result object low. And I looked through all courses that we had in the courses in the last three years and no exercise had this much tests.
But the code accounts for this problem and limits the range of the value in the server (see setters) and client (see result.service.ts where we show "127+" in these cases) as well.

Reply:
Additionally tinyInt is unsigned, so we actually have 255 as the maximum value

Reply:
Okay, if the code handles this limitation properly, then it is perfectly fine.

Are you sure tinyints are always unsigned? The [MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html) seems to suggest both unsigned and signed are possible:
- ‘Numeric data types that permit the UNSIGNED attribute also permit SIGNED. However, these data types are signed by default, so the SIGNED attribute has no effect.’
- ‘`TINYINT[(M)] [UNSIGNED] [ZEROFILL]`: A very small integer. The signed range is -128 to 127. The unsigned range is 0 to 255.’

Reply:
I found another documentation that only mentioned the range 0-255.
No it's explicitly stated that the column is unsigned